lib/matplotlib/artist.py

```diff
@@ -461,28 +461,18 @@ def get_children(self):
         r"""Return a list of the child `.Artist`\s of this `.Artist`."""
         return []
 
-    def _default_contains(self, mouseevent, figure=None):
+    def _not_same_canvas(self, event):
         """
-        Base impl. for checking whether a mouseevent happened in an artist.
+        Check whether an *event* occurred on a canvas other that this artist's canvas.
 
-        1. If the artist figure is known and the event did not occur in that
-           figure (by checking its ``canvas`` attribute), reject it.
-        2. Otherwise, return `None, {}`, indicating that the subclass'
-           implementation should be used.
+        Subclasses should start their definition of `contains` as follows::
 
-        Subclasses should start their definition of `contains` as follows:
-
-            inside, info = self._default_contains(mouseevent)
-            if inside is not None:
-                return inside, info
+            if self._not_same_canvas(mouseevent):
+                return False, {}
             # subclass-specific implementation follows
-
-        The *figure* kwarg is provided for the implementation of
-        `.Figure.contains`.
         """
-        if figure is not None and mouseevent.canvas is not figure.canvas:
-            return False, {}
-        return None, {}
+        return (getattr(event, "canvas", None) is not None and self.figure is not None
+                and event.canvas is not self.figure.canvas)
```

Comment:
Rather than guarding against the `None is None` case, could you do something like this instead?
```suggestion
        return (getattr(event, "canvas", object()) is not getattr(self.figure, "canvas", object())
```

Reply:
I don't think so?  That would incorrectly return "different" if event.canvas = None and self.figure != None and self.figure.canvas = anything, I believe?

Reply:
Ahh, yes. My confusion seems to agree with the other confusion on the "maybe"/"for sure" same canvas comments above.

I wonder if there is a way to clear up that maybe_same_canvas by guaranteeing these attributes are set elsewhere in another PR.